### PR TITLE
Support Python 3.13

### PR DESCRIPTION
**Pull request by jonadaly**
Hi @bodono , I tried to mirror [your changes](https://github.com/bodono/scs-python/pull/86) from my previous request to support Python 3.12 / MacOS - now for the new release of Python 3.13!

I've also removed support for Python 3.8 since it's now [end-of-life](https://devguide.python.org/versions/).

### CODE REVIEW

**Comment by bodono** [comment]
Thanks for looking into this @jonadaly ! It looks like one the python 3.9 run on windows is failing, possibly due to an incompatible numpy version, do you have any ideas on how to fix that? Once fixed we can land this!

**Comment by jonadaly** [comment]
I'm not sure why that's happening - what's weird is that fixing it seems to have broken the 3.12 build! 😆

**Comment by enzbus** [comment]
@jonadaly @bodono  maybe try to remove the constraint on setuptools here -> https://github.com/bodono/scs-python/blob/d4f99f9bc4d2b6395d81a66a8cd298fd188a6da4/.github/workflows/build.yml#L46 , dependencies may have broken in Conda (version 60 is 3 years old).

**Comment by bodono** [comment]
> I'm not sure why that's happening - what's weird is that fixing it seems to have broken the 3.12 build! 😆

We could drop support for 3.9 too, that's less than a year from EOL, if we do that and revert back to when 3.12 was working we can land this.

**Comment by bodono** [comment]
And sorry for the delay in responding! Thanks for putting effort into this.

**Comment by jonadaly** [comment]
So I've got the CI passing by removing the "legacy build" for >3.11. Not sure how legit that is though - what do you think @bodono ?

File: .github/workflows/build.yml
Comment (bodono) [inline .github/workflows/build.yml:159]: If this is just always False then we can just remove it, and from line 175 below.

**Review by bodono — CHANGES_REQUESTED** [review]

**Comment by bodono** [comment]
Thanks @jonadaly, LGTM after minor update I suggested. Let me know when you think it is ready to merge and I can do it.

**Review by bodono — APPROVED** [review]

**Comment by bodono** [comment]
Thanks!

**Pull request merged**
